modules/core/src/main/java/org/apache/ignite/internal/processors/cache/IgniteCacheOffheapManagerImpl.java

```diff
@@ -3074,12 +3074,13 @@ else if (tombstoneRow != null) {
                     replicationRequire(tombstoneRow.version()))
                     addUpdateToLog(new UpdateLogRow(cctx.cacheId(), tombstoneRow.version().updateCounter(), tombstoneRow.link()));
 
-                if (oldRow != null && oldRow.version().updateCounter() != 0) {
-                    if (oldTombstone && tombstoneRow == null)
-                        cctx.dr().onTombstoneCleaned(partId, oldRow.version().updateCounter());
+            }
 
-                    removeFromLog(new UpdateLogRow(cctx.cacheId(), oldRow.version().updateCounter(), oldRow.link()));
-                }
+            if (oldRow != null && oldRow.version().updateCounter() != 0) {
+                if (oldTombstone && tombstoneRow == null)
+                    cctx.dr().onTombstoneCleaned(partId, oldRow.version().updateCounter());
```

Comment:
```suggestion
                if (isIncrementalDrEnabled(cctx) && oldTombstone && tombstoneRow == null)
                    cctx.dr().onTombstoneCleaned(partId, oldRow.version().updateCounter());
```
Otherwise, if DR is disabled, it may lead to unexpected behaviour or NPE.

Reply:
Thank you!